src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -552,6 +554,8 @@ handle_exception (MonoException *exc, MonoContext *throw_ctx, MonoContext *catch
 {
 	ERROR_DECL (error);
 	PRINT_DEBUG_MSG (1, "handle exception - %d - %p - %p - %p\n", pause_on_exc, exc, throw_ctx, catch_ctx);
+	
+	exception_on_runtime_invoke = exc;
```

Comment:
Maybe add a comment here from your PR summary: normal mono_runtime_try_invoke does not capture the exception and this is a temporary workaround.